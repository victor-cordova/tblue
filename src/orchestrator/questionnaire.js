// @flow
import inquirer from 'inquirer';
import logger from '~/common/logger';
import templates from '../../templates';
import type { Questionnaire } from '~/types';

const questionName = 'selectedQuestionnaire';

export const solicitQuestionnaire = async (): Questionnaire => {
    logger.instructions(
        'Questionnaires contain pre-defined cards and also cards generated based on your answers');
    const answers = await inquirer
        .prompt([{
            type: 'list',
            name: questionName,
            message: 'What questionnaire would you like to use?',
            choices: templates.map(q => q.name),
            prefix: 'ðŸ“‹',
        }]);
    return templates.find(q => q.name === answers[questionName]);
};
