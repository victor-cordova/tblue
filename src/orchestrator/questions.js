import inquirer from 'inquirer';

import type { Card, Blueprint } from '~/types';

export const askQuestions = async (blueprint: Blueprint): Array<Card> => {
    const { questions } = blueprint;
    const promptQuestions = questions.map(
        question => ({
            type: 'confirm',
            name: question.content,
            message: question.content,
        })
    );
    const answers = await inquirer.prompt(promptQuestions);
    return questions
        .filter(question => answers[question.content])
        .map(question => question.cardToGenerate);
};
