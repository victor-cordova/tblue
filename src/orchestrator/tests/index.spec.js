jest.mock('../questionnaire');
jest.mock('../questions');

import { solicitQuestionnaire } from '../questionnaire';
import { askQuestions } from '../questions';
import { begin } from '../index';
import { basicQuestionnaire, questionGeneratedCards } from './stubs';

describe('orchestrator', () => {
    describe('begin', () => {
        it('returns correct cards', async () => {
            solicitQuestionnaire.mockImplementation(() => basicQuestionnaire);
            askQuestions.mockImplementation(() => questionGeneratedCards);
            expect(await begin()).toMatchSnapshot();
        });
    });
});