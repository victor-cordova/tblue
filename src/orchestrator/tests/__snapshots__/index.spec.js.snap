// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`orchestrator begin returns correct cards 1`] = `
Array [
  Object {
    "name": "Create dockerfile",
  },
  Object {
    "name": "Add google analytics",
  },
  Array [
    Object {
      "description": "- Create repo in ",
      "name": "Repository configuration",
    },
    Object {
      "description": "",
      "name": "Configure react router",
    },
    Object {
      "name": "Configure jest",
    },
    Object {
      "name": "Configure babel root import",
    },
    Object {
      "name": "Add pre push hooks",
    },
    Object {
      "description": "",
      "name": "Define folder structure",
    },
    Object {
      "description": "",
      "name": "Configure dev environment",
    },
    Object {
      "description": "",
      "name": "Configure CI",
    },
  ],
]
`;
